### ISSUE TITLE
Inspection mode for debugging and diagnostics

### ISSUE THREAD

**Issue by cdata** [google/model-viewer#218]
It would be useful to be able to force any given `<model-viewer>` into an inspection mode when debugging. This could be as simple as having the element dump information to the console.

**Comment by theprojectsomething** [comment]
Would be great to have some visual helpers available as well, e.g:

- bounding boxes for models
- bounding spheres (max and min), current coords / normals for camera orbit (imo more useful than model BBs)
- planes for annotations

```
<model-viewer data-debug="camera model">
  <button slot="hotspot-0" data-debug></button>
</model-viewer>
```

**Comment by elalish** [comment]
@theprojectsomething I like your example! It's just the two of us on this project and we have a ton of work to do; if you'd be interested in contributing a PR for this, we'd love to work with you!

**Comment by theprojectsomething** [comment]
@elalish absolutely, happy to contribute a PR. Will try to find some time in the next few weeks to go over the code base properly. Cheers to both of you for all your work!

**Comment by theprojectsomething** [comment]
I have zero typescript so have put together add-on package instead, for inclusion in development environments: https://www.npmjs.com/package/model-viewer-debugger

**Comment by elalish** [comment]
@theprojectsomething Awesome, thank you! We're busy upgrading our docs, so we should link to this as we stabilize them.

**Comment by cdata** [comment]
@theprojectsomething That's awesome! You might find the debug mode we added in https://github.com/google/model-viewer/pull/856 (and released in v0.8.0) to be useful.

Basically it gives you access to the internal renderer and a few critical Three.js constructors from the version of Three.js that is bundled w/ `<model-viewer>`.

**Comment by theprojectsomething** [comment]
Ah great, thanks @cdata ... as you can see I opted for research-lite / brute force 👍💣 Will refactor with these in mind as soon as I get a chance.
